kele.el

```diff
@@ -1106,10 +1123,139 @@ Only populated if Embark is installed.")
       (kele--disable)
     (kele--enable)))
 
+(defun kele--namespace-read (prompt initial-input history)
+  "Read a namespace value using PROMPT, INITIAL-INPUT, and HISTORY.
+
+Assumes that the current Transient prefix's :scope is an alist w/ `context' key."
+  ;; FIXME: Make this resilient to the prefix's scope not having a context
+  ;; value.  If not present (or the scope is not an alist or the scope is not
+  ;; defined), default to current context.
+  (if-let ((context (alist-get 'context (oref transient--prefix scope))))
+      (completing-read
+       prompt
+       (-cut kele--resources-complete <> <> <>
+             :cands (kele--get-namespaces
+                     (alist-get 'context (oref transient--prefix scope)))
+             :category 'kele-namespace)
+       nil t initial-input history)
+    (error "Unexpected nil context in `%s'" (oref transient--prefix command))))
+
+(defclass kele--transient-scope-modifier (transient-option)
+  ((scope-key
+    :initarg :scope-key
```

Comment:
Maybe instead of basing this around `:scope-key` -- implicitly requiring that the prefix's scope value be an alist -- we can simply have the infix take in a `(lambda (scope new-value)` that acts on the scope via argument.

Reply:
Done